drivers/telescope/rainbow.cpp

```diff
@@ -230,6 +241,17 @@ bool Rainbow::ISNewSwitch(const char *dev, const char *name, ISState *states, ch
             IDSetSwitch(&HomeSP, nullptr);
             return true;
         }
+        else if (!strcmp(SaveAlignBeforeSyncSP.name, name)) {
+            if (SaveAlignBeforeSyncS[0].s == ISS_ON)
```

Comment:
No need to do this, just do IUUpdateSwitch(...) then SaveAlignBeforeSyncSP.s = IPS_OK, then setIDSetSwitch(&Save...)

Reply:
Thanks I have made the changes to my latest push